compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -786,8 +786,24 @@ pub trait PrintState<'a>: std::ops::Deref<Target = pp::Printer> + std::ops::Dere
             token::CloseDelim(Delimiter::Bracket) => "]".into(),
             token::OpenDelim(Delimiter::Brace) => "{".into(),
             token::CloseDelim(Delimiter::Brace) => "}".into(),
-            token::OpenDelim(Delimiter::Invisible) => "/*«*/".into(),
-            token::CloseDelim(Delimiter::Invisible) => "/*»*/".into(),
+            // We use comment-delimited phrases here for invisible delimiters,
+            // because these cases tend to occur in error messages. In
+            // `print_mac_common` we use `/*«*/` and `/*»*/` for compactness,
+            // because that occurs in other cases like pretty-printing.
+            token::OpenDelim(Delimiter::Invisible(src)) => match src {
+                InvisibleSource::ExprMv => "/*start of expr expansion*/",
+                InvisibleSource::FlattenNonterminals | InvisibleSource::ProcMacro => {
+                    "/*invisible open delimiter*/"
+                }
+            }
+            .into(),
+            token::CloseDelim(Delimiter::Invisible(src)) => match src {
+                InvisibleSource::ExprMv => "/*end of expr expansion*/",
+                InvisibleSource::FlattenNonterminals | InvisibleSource::ProcMacro => {
+                    "/*invisible close delimiter*/"
+                }
+            }
+            .into(),
```

Comment:
Parser uses `fn token_descr(_opt)` for printing expected/found tokens, so it can also be used as a customization point.
(You won't need to wrap anything in comments there, for example.)